tools/reviewer/web/src/app/dashboard/reviews-panel/review/review.component.ts

```diff
@@ -35,14 +41,36 @@ export class ReviewComponent implements OnInit {
     });
   }
 
+  // Upon click on a file open a
+  // single file reivew page showing
+  // code difference and comments
   openFile(filePosition): void {
-    this.zone.run(() => {
-      // Build a route path on the following format
-      // /diff/<diff number>/<path>?
-      // ls=<left snapshot number>&rs=<right snapshot number>
-      this.router.navigate(['diff/' + this.diffId + '/' + filePosition], {
-        queryParams: { ls: '1', rs: '3' }
-      });
+    // Build a route path on the following format
+    // /diff/<diff number>/<path>?
+    // ls=<left snapshot number>&rs=<right snapshot number>
+    this.router.navigate(['diff/' + this.diffId + '/' + filePosition], {
+      queryParams: { ls: '1', rs: '3' }
     });
   }
+
+  // Update the Diff in the DB
+  updateDiff(diff: Diff, message: string): void {
+    this.firebaseService
+      .updateDiff(diff)
+      .then(res => {
+        // TODO make separate service for notifications
```

Comment:
Looks like you have a `NotificationService`, no?

Reply:
yep.